rules/CVE-2022-3172/raw.rego

```diff
@@ -3,49 +3,62 @@ package armo_builtins
 import future.keywords.in
 
 deny[msg] {
+	# find aggregated API APIServices
 	obj = input[_]
 	obj.apiVersion == "apiregistration.k8s.io/v1"
 	obj.kind == "APIService"
-	service := obj.spec.service
+	api_service := obj.spec.service
 
+	# check API server version vulnerability
 	api_infos = [api_info |
 		api_info := input[i]
 		api_info.apiVersion == "apiserverinfo.kubescape.cloud/v1beta0"
 		api_info.kind == "APIServerInfo"
 		api_info.metadata.name == "version"
 	]
 
-    version = get_api_server_version(api_infos)
-    is_api_server_version_affected(version)
+	version = get_api_server_version(api_infos)
+	is_api_server_version_affected(version)
+
+	# Find the service that exposes the extended API
+	services = [obj |
+		obj := input[j]
+		obj.apiVersion == "v1"
+		obj.kind == "Service"
+		obj.metadata.name == api_service.name
+	]
+
+	count(services) == 1
+	service = services[0]
 
 	msg := {
-		"alertMessage": "the following APIService may redirect client traffic to any URL",
+		"alertMessage": "the following pair of APIService and Service may redirect client traffic to any URL",
 		"alertScore": 2,
 		"failedPaths": [],
 		"fixPaths": [],
 		"packagename": "armo_builtins",
-		"alertObject": {"k8sApiObjects": [obj]},
+		"alertObject": {"k8sApiObjects": [obj, service]},
```

Comment:
please change alert object from k8sApiObjects to externalObjects where 'obj' is the main object and 'service' is the related obj

Reply:
For now, it seems that Kubescape does not support this kind of `relatedObjects` when it's not a RoleBinding.